File: addon.cc
#include <node.h>
#include "routines.h"

void Init(v8::Local<v8::Object> exports) {
  NODE_SET_METHOD(exports, "sasum", sasum);
  NODE_SET_METHOD(exports, "dasum", dasum);
}

NODE_MODULE(addon, Init)

File: addon.js
(function () {
  'use strict';

  var addon = require('./build/Release/addon.node');

  var x = new Float32Array([1, 2, 3]),
      y = new Float32Array([3, 2, 1]);

  console.log(addon.sasum(3, x, 1));

  addon.saxpy(3, 1, x, 1, y, 1);
  console.log(y);
}());
